src/index.js

```diff
@@ -61,239 +61,177 @@ class Radar {
 
   static getLocation(callback) {
     if (!callback) {
-      throw new Error(STATUS.ERROR_PARAMETERS);
+      throw new Error(ERROR.MISSING_CALLBACK);
     }
 
-    Navigator.getCurrentPosition((status, { latitude, longitude, accuracy }) => {
-      if (status !== STATUS.SUCCESS) {
-        callback(status);
-        return;
-      }
-
-      callback(status, { latitude, longitude, accuracy });
-      return;
-    });
+    Navigator.getCurrentPosition()
+      .then((location) => {
+        callback(null, location);
+      })
+      .catch(callback);
   }
 
   static trackOnce(arg0, arg1) {
-    if (!arg0 || (typeof arg0 === 'function')) {
-      const callback = arg0;
-      Track.trackOnce(
-        (status, location, user, events) => {
-          if (callback) {
-            callback(status, location, user, events);
-            return;
-          }
-        }
-      )
+    if (!arg0) {
+      throw new Error(ERROR.MISSING_CALLBACK);
+    }
+
+    let callback;
+    let position;
+
+    if (typeof arg0 === 'function') {
+      callback = arg0;
+
     } else if (typeof arg0 === 'object') {
-      const { latitude, longitude, accuracy } = arg0;
-      const callback = arg1;
-      Track.trackOnceWithLocation({ latitude, longitude, accuracy },
-        (status, location, user, events) => {
-          if (callback) {
-            callback(status, location, user, events);
-            return;
-          }
-        }
-      )
+      position = arg0;
+
+      if (typeof arg1 !== 'function') {
+        throw new Error(ERROR.MISSING_CALLBACK);
+      }
+      callback = arg1;
+
     } else {
-      throw new Error(STATUS.ERROR_PARAMETERS);
+      throw new Error(ERROR.PARAMETERS);
     }
+
+    Track.trackOnce(position)
+      .then((response) => {
+        const { location, user, events } = response;
+        callback(null, location, user, events, response);
+      })
+      .catch(callback);
   }
 
   static getContext(arg0, arg1) {
     if (!arg0) {
-      throw new Error(STATUS.ERROR_PARAMETERS);
+      throw new Error(ERROR.MISSING_CALLBACK);
```

Comment:
will this potentially be an issue if someone does `util.promisify(getContext)`? then, the call would look like `await Radar.getContext()`, so technically no `arg0`

Reply:
I'm not sure about the internals, but I suspect the promisify wrapper will be giving this a callback (this code will not work without a callback)